packages/frontend/src/components/common/modal/ChartCreateModal/SaveToSpaceOrDashboard.tsx

```diff
@@ -0,0 +1,428 @@
+            />
+            <Button
+                size="xs"
+                variant="default"
+                mr="auto"
+                compact
+                leftIcon={<MantineIcon icon={IconPlus} />}
+                onClick={() => setShouldCreateNewSpace(true)}
+            >
+                Create new space
+            </Button>
+        </Stack>
+    );
+};
+
+type SaveToDashboardProps = Pick<SaveToSpaceProps, 'form' | 'spaces'> & {
+    dashboards: DashboardBasicDetails[] | undefined;
+    isLoadingDashboards: boolean;
+};
+
+const SaveToDashboard: FC<SaveToDashboardProps> = ({
+    form,
+    spaces,
+    dashboards,
+    isLoadingDashboards,
+}) => {
+    if (!dashboards) return null;
+
+    return (
+        <Select
+            description="Select a dashboard to save the chart directly to"
+            id="select-dashboard"
+            label="Dashboard"
+            size="xs"
+            data={dashboards.map((d) => ({
+                value: d.uuid,
+                label: d.name,
+                group: (spaces ?? []).find((s) => s.uuid === d.spaceUuid)?.name,
+            }))}
+            rightSection={isLoadingDashboards && <Loader size="xs" />}
+            defaultValue={
+                dashboards.find((d) => d.spaceUuid === form.values.spaceUuid)
+                    ?.uuid
+            }
+            searchable
+            nothingFound="No matching dashboards found"
+            filter={(value, dashboard) =>
+                !!dashboard.label
+                    ?.toLowerCase()
+                    .includes(value.toLowerCase().trim())
+            }
+            withinPortal
+            required={form.values.saveDestination === SaveDestination.Dashboard}
+            {...form.getInputProps('dashboardUuid')}
+        />
+    );
+};
+
+type SaveToSpaceOrDashboardProps = {
+    savedData: CreateSavedChartVersion;
+    defaultSpaceUuid: string | undefined;
+    dashboardInfoFromSavedData: {
+        dashboardUuid: string | null;
+        dashboardName: string | null;
+    };
+    onConfirm: (savedData: CreateSavedChartVersion) => void;
+    onClose: () => void;
+};
+
+export const SaveToSpaceOrDashboard: FC<SaveToSpaceOrDashboardProps> = ({
+    savedData,
+    defaultSpaceUuid,
+    dashboardInfoFromSavedData,
+    onConfirm,
+    onClose,
+}) => {
+    const { projectUuid } = useParams<{ projectUuid: string }>();
+
+    const { mutateAsync: createChart } = useCreateMutation();
+
+    const [, setShouldCreateNewSpace] = useState(false);
+    const { mutateAsync: createSpace } = useSpaceCreateMutation(projectUuid);
+
+    const form = useForm<ChartCreateModalFormValues>({
+        initialValues: {
+            name: '',
+            spaceUuid: '',
+            dashboardUuid: '',
+            dashboardName: '',
+            description: '',
+            newSpaceName: null,
+            saveDestination: SaveDestination.Space,
+        },
```

Comment:
something that I've learned recently...

if initial state depends on the remote call - you can actually ommit initialValues and once the data loads you can call `form.initialize({ initialstategoeshere })`

you can distinguish between initialized/un-initialized form by using `form.initialized` property

Reply:
This is so good! thank you!!